woocommerce/woocommerce-android
# Issue/12004 upload selected photo

Closes: #12004


### Description

This PR adds the following: 
- Handles image uploading if there's an image selected
- Handle errors uploading images
- Prepared the logic to handle UI changes when saving the product as a draft or publishing the product


### Testing information
1. Log into an atomic site or site with Jetpack AI-enabled
2. Navigate to the products tab
3. Add product using AI option
4. Select an image that has not yet being uploaded to WP media library
5. Click Generate product details
6. From preview screen click on "Save as draft" 
7. Notice how the spinner is shown while the image is uploaded
8. Navigate back and chick on replace image and selected WP media library. Check the image is now uploaded to the library
9. Repetar steps 5 to 7 but this time disable internet connection before clicking on "Save as draft". Check the error dialog is shown. 

### Images/gif


https://github.com/user-attachments/assets/2624d462-c8d9-4522-af95-eda7af683b0e


https://github.com/user-attachments/assets/502cc1c8-13ca-4c12-b75e-b4e586ffb155
